core/src/main/java/org/elasticsearch/action/admin/indices/template/put/PutIndexTemplateRequest.java

```diff
@@ -290,16 +293,17 @@ public PutIndexTemplateRequest source(Map templateSource) {
             String name = entry.getKey();
             if (name.equals("template")) {
                 if(entry.getValue() instanceof String) {
-                    template(Collections.singletonList((String) entry.getValue()));
+                    patterns(Collections.singletonList((String) entry.getValue()));
                 } else if (entry.getValue() instanceof List) {
-                    template((List<String>) entry.getValue());
+                    List<String> elements = ((List<?>) entry.getValue()).stream().map(Object::toString).collect(Collectors.toList());
+                    patterns(elements);
                 } else {
-                    throw new IllegalArgumentException("Malformed [template] value, should be a string[]");
+                    throw new IllegalArgumentException("Malformed [template] value, should be a string or a list of strings");
                 }
             } else if (name.equals("order")) {
                 order(XContentMapValues.nodeIntegerValue(entry.getValue(), order()));
             } else if ("version".equals(name)) {
-                if ((entry.getValue() instanceof Integer) == false) {
+                if (!(entry.getValue() instanceof Integer)) {
```

Comment:
We actually prefer `== false` or `false ==` of `!`. Well, we have one core contributor who **really** prefers it and the rest of us are fine with it. `false ==` is unlikely to be misread, whereas `!` is small and can get lost if you scan the code quickly. So I'd change this back. It is weird, yes, but it is our kind of weird.